src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -4062,40 +4062,8 @@ public bool IsReplicationCertificate(TransactionOperationContext context, string
                 new Span<byte>(p, size).CopyTo(buffer);
                 using var knownCert = new X509Certificate2(buffer);
 
-                var userChain = new X509Chain();
-                var knownCertChain = new X509Chain();
-                try
-                {
-                    userChain.Build(userCert);
-                }
-                catch (Exception)
-                {
-                    return false;
-                }
-
-                try
-                {
-                    knownCertChain.Build(knownCert);
-                }
-                catch (Exception)
-                {
-                    return false;
-                }
-
-                if (knownCertChain.ChainElements.Count != userChain.ChainElements.Count)
-                {
-                    return false;
-                }
-
-                for (var i = 0; i < knownCertChain.ChainElements.Count; i++)
-                {
-                    // We walk the chain and compare the user certificate vs one of the existing certificate with same pinning hash
-                    var currentElementPinningHash = userChain.ChainElements[i].Certificate.GetPublicKeyPinningHash();
-                    if (currentElementPinningHash != knownCertChain.ChainElements[i].Certificate.GetPublicKeyPinningHash())
-                    {
-                        return false;
-                    }
-                }
+                if (CertificateUtils.CertHasKnownIssuer(userCert, knownCert, securityConfiguration, out var _) == false)
```

Comment:
Couple of things that I'm not sure about:
1. We are iterating here through a list of certificates, but old code when there was an exception or in few other places was returning immediately `false`. Shouldn't we go over the whole list anyway?
2. I've moved `CertHasKnownIssuer` to CertificateUtils, but it know also checks WellKnownIssuerHashes and we were not doing it here. Is this correct?

CC @ayende @iftahbe

Reply:
1. For the pinning hash to be accepted, we compare the entire chain, and if even one element is different we return false and don't trust it. No point in checking the next elements.

Reply:
2. I think you are correct and it should have also checked the WellKnownIssuerHashes. It probably was a mistake.